backend/src/users/user-relationship/user-relationship.service.ts

```diff
@@ -14,29 +14,38 @@ export class UserRelationshipService extends AbstractService<UserRelationship> {
 		super(repository);
 	}
 	//  check that id: a - b and id: b - a doesn't already exists
-	async hasExistingRelationship(
-		relationshipRequest: CreateUserRelationshipDto,
-	): Promise<boolean> {
-		const relationships: UserRelationship[] = await this.findAll({
-			relations: {
-				source_id: true,
-				target_id: true,
-			},
-			where: {
-				source_id: {
-					id: In([
-						relationshipRequest.source_id,
-						relationshipRequest.target_id,
-					]),
+	async getExistingRelationship(
+		source: number,
+		target: number,
+	): Promise<UserRelationship> | null {
+		try {
+			const relationship: UserRelationship = await this.findOne({
+				relations: {
+					source_id: true,
+					target_id: true,
 				},
-				target_id: {
-					id: In([
-						relationshipRequest.source_id,
-						relationshipRequest.target_id,
-					]),
+				where: {
+					source_id: {
+						id: In([source, target]),
+					},
+					target_id: {
+						id: In([source, target]),
+					},
 				},
-			},
-		});
-		return relationships.length > 0;
+			});
+			return relationship;
+		} catch (e) {
+			return null;
+		}
+	}
+
+	async hasExistingRelationship(
```

Comment:
just to clarify: so we would probably always use hasExistingRealtionship to protect the get search??

Reply:
This is/was for the post request, but the @post controller wont be used anymore when I continue with the profileList ticket, as all users will be instanciated with a relationship of NONE so there'll always be a relationship